pkg/cri/sbserver/update_runtime_config.go

```diff
@@ -138,3 +120,35 @@ func getRoutes(cidrs []string) ([]string, error) {
 	}
 	return routes, nil
 }
+
+func writeCNIConfigFile(ctx context.Context, confDir string, confTemplate string, podCIDR string, podCIDRRanges []string, routes []string) error {
+	log.G(ctx).Infof("Generating cni config from template %q", confTemplate)
+	// generate cni config file from the template with updated pod cidr.
+	t, err := template.ParseFiles(confTemplate)
+	if err != nil {
+		return fmt.Errorf("failed to parse cni config template %q: %w", confTemplate, err)
+	}
+	if err := os.MkdirAll(confDir, 0755); err != nil {
+		return fmt.Errorf("failed to create cni config directory: %q: %w", confDir, err)
+	}
+	f, err := os.CreateTemp(confDir, cniConfigFileName)
+	if err != nil {
+		return fmt.Errorf("failed to create temp config file: %w", err)
+	}
+	defer f.Close()
+	if err := t.Execute(f, cniConfigTemplate{
+		PodCIDR:       podCIDR,
+		PodCIDRRanges: podCIDRRanges,
+		Routes:        routes,
+	}); err != nil {
+		return fmt.Errorf("failed to generate cni config file %q: %w", f.Name(), err)
+	}
+	if err := f.Sync(); err != nil {
+		return fmt.Errorf("failed to fsync cni config file %q: %w", f.Name(), err)
+	}
+	confFile := filepath.Join(confDir, cniConfigFileName)
+	if err := os.Rename(f.Name(), confFile); err != nil {
+		return fmt.Errorf("failed to move generated cni config file from %q to %q: %w", f.Name(), confFile, err)
```

Comment:
The old file perms were 0644, so we'd lose read for group and world with this (CreateTemp uses 0600), any concerns?

Reply:
This is a good call-out.  We can reset the permissions back to 0644 before renaming.